src/EnergyPlus/SolarShading.cc

```diff
@@ -4970,7 +4970,7 @@ void FigureSolarBeamAtTimestep(EnergyPlusData &state, int const iHour, int const
 
                 for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
 
-                    if (!state.dataSurface->Surface(SurfNum).ShadowingSurf &&
+                    if (!state.dataSurface->SurfIsShadowing(SurfNum) &&
                         (!state.dataSurface->Surface(SurfNum).HeatTransSurf || !state.dataSurface->Surface(SurfNum).ExtSolar))
```

Comment:
Why is `IsShadowing` a surface-level array while `HeatTransSurf` and `ExtSolar` are fields in the Surface array? It seems like these fields have similar usages.

Reply:
@amirroth `HeatTransSurf` and `ExtSolar` are assigned during input processing and saved in `SurfaceTmp` (e.g.`ExtSolar  = UtilityRoutines::SameString(state.dataIPShortCut->cAlphaArgs(ArgPointer), "SunExposed")`). With all surface data being read, `SurfaceTmp` arrays are sorted and copied to a new `Surface` array. `IsShadowing` is evaluated and assigned afterward. I don't know if it's worthwhile or good practice to move it out from a field to an array.

Reply:
I am not suggesting removing it from the class, I was suggesting replicating it in its own array if the use cases warrants it. If not (maybe surfaces are sorted so that you don't have to explicitly check for these conditions or maybe we are using explicit surface lists that only include surfaces with these conditions) then it doesn't matter.